src/psyclone/gocean1p0.py

```diff
@@ -2535,6 +2535,13 @@ def psyir_expression(self):
         if six.text_type(self.name).isnumeric():
             return Literal(self.name, INTEGER_TYPE)
 
+        # Now try for a real value:
+        try:
+            float(self.name)
```

Comment:
This is a good idea but since I am not sure we can guarantee that python floats syntax always match with PSyIR syntax, could we maybe use `if re.match(Literal._real_value, self.name):`?

Reply:
Even better - I just realised that we can remove the use of `float`, since the constructor for `Literal` will raise an exception. I've documented this, it is indeed nicer.